### PR TITLE
added map to sidebar elements

### PR DESCRIPTION
**Pull request by vanshika703**

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+  function handleToggleButton(){
+    console.log("toggleMenu",toggleMenu)
+    dispatch(toggleMenu())
+    console.log("dispatched toggle");
+  }
   return (
     <div className="grid grid-flow-col my-2 p-2 shadow-md">
       <div className="flex col-span-1">
         <img
+          onClick={()=>handleToggleButton()}
```
Comment (harshgoel05) [inline src/components/Header.js]: onClick={handleToggleButton}

File: src/components/Sidebar.js [hunk 2]
```diff
@@ -1,26 +1,223 @@
...
 
   return (
-    <div className='col-span-1 text-left m-2 p-2'>
-      <ul className='my-4'>
-        {
-          sidebarItems.map((sidebarItem) => <li className='p-2'>{sidebarItem}</li>)
-        }
+    <div className="col-span-1 text-left m-2 p-2">
+      <ul className="my-4">
+        {sidebarItems.map((sidebarItem) => (
```
Comment (harshgoel05) [inline src/components/Sidebar.js]:
```suggestion
        {sidebarItems && sidebarItems.map((sidebarItem) => (
```

Comment (harshgoel05) [reply src/components/Sidebar.js]: Always check sidebarItems is null or not

File: src/components/Sidebar.js [hunk 3]
```diff
@@ -1,26 +1,223 @@
...
     </div>
-  )
-}
+  );
+};
 
-export default Sidebar;
\ No newline at end of file
+export default Sidebar;
+/*   let c = "block";
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: Delete all comments

File: src/components/Header.js [hunk 4]
```diff
@@ -1,10 +1,18 @@
-
+import {useDispatch} from "react-redux"
+import {toggleMenu} from "../utils/toggleSlice"
 
 const Header = () => {
+  const dispatch = useDispatch();
+  function handleToggleButton(){
+    console.log("toggleMenu",toggleMenu)
```
Comment (harshgoel05) [inline src/components/Header.js]: Remove all logs

File: src/components/Sidebar.js [hunk 5]
```diff
@@ -1,26 +1,223 @@
-import React from 'react'
+import React from "react";
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: This can be omitted, not sure

**Review by harshgoel05 — CHANGES_REQUESTED** [review]

File: src/components/ButtonBar.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
 import React from "react";
 
 const ButtonBar = () => {
-    const categories = ["All", "Mixes", "Music", "Comedy", "ReactJS", "Standup", "Cricket", "Mixes", "Cricket", "Mixes", "WebDev", "Travel", "Standup",]
+    const categories = ["All", "Mixes", "Music", "Comedy", "ReactJS", "Standup", "Cricket", "Mixes", "Mixes", "WebDev", "Travel", "Standup",]
   return (
     <div>
         {
-            categories.map((category) => <button className="py-1 px-4 m-3 bg-stone-100 rounded-lg">{category}</button>)
+            categories.map((category, index) => <button key={index} className="py-1 px-4 m-3 bg-stone-100 rounded-lg">{category}</button>)
```
Comment (harshgoel05) [inline src/components/ButtonBar.js]: no index

File: src/components/Header.js [hunk 7]
```diff
@@ -1,10 +1,39 @@
...
+  }
+
+  async function getSeachSuggestions() {
+    console.log("search", searchQuery);
+    const data = await fetch(YOUTUBE_SEARCH_API + searchQuery);
+    const json = await data.json();
+    setSuggestions(json[1]);
+  }
+
+  useEffect(() => {
```
Comment (harshgoel05) [inline src/components/Header.js:28]: Try debouncing with [this](https://dmitripavlutin.com/react-throttle-debounce/)

File: src/components/Header.js [hunk 8]
```diff
@@ -15,16 +44,38 @@ const Header = () => {
...
+              alt="search"
+              className="h-3"
+              src="https://cdn-icons-png.flaticon.com/512/3917/3917132.png"
+            />
+          </button>
+        </div>
+        {showSuggestions && (
+          <div className="fixed bg-white w-1/4 p-2 mx-5 border border-gray-200 rounded-lg shadow-lg">
+            <ul>
+              {suggestions.map((suggestion, index) => (
```
Comment (harshgoel05) [inline src/components/Header.js]: no index

File: src/components/Sidebar.js [hunk 9]
```diff
@@ -1,26 +1,223 @@
...
+import { useSelector } from "react-redux";
 
 const Sidebar = () => {
+  const isMenuOpen = useSelector((store) => store.toggle.isMenuOpen);
+  
+  const sidebarItems = [
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: move this to a new file in constants folder

File: src/components/Sidebar.js [hunk 10]
```diff
@@ -1,26 +1,223 @@
...
   return (
-    <div className='col-span-1 text-left m-2 p-2'>
-      <ul className='my-4'>
-        {
-          sidebarItems.map((sidebarItem) => <li className='p-2'>{sidebarItem}</li>)
-        }
+    <div className="col-span-1 text-left m-2 p-2">
+      <ul className="my-4">
+        {sidebarItems.map((sidebarItem, index) => (
+          <li key={index} className="p-2">
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: no index

File: src/components/Sidebar.js [hunk 11]
```diff
@@ -1,26 +1,223 @@
...
+            <div className="flex">
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: use any one of justify-center items-center to center, according to use case
```suggestion
            <div className="flex justify-center items-center">
```

File: src/components/Sidebar.js [hunk 12]
```diff
@@ -1,26 +1,223 @@
...
+              <img alt="icon" className="h-5 m-1" src={sidebarItem.icon} />
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: give proper alt for all images, this is staic rn, should be dyanmiuc

File: src/components/Sidebar.js [hunk 13]
```diff
@@ -1,26 +1,223 @@
...
+            </div>
+          </li>
+        ))}
       </ul>
       <hr></hr>
-      <ul className='my-4'>
-        {
-          sidebarItems2.map((sidebarItem) => <li className='p-2'>{sidebarItem}</li>)
-        }
+      {/* <ul className="my-4">
```
Comment (harshgoel05) [inline src/components/Sidebar.js]: Remove comments

File: src/utils/toggleSlice.js [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const toggleSlice = createSlice({
+  name: "toggle",
+  initialState: {
+    isMenuOpen: true,
+  },
+  reducers: {
+    toggleMenu: (state) => {
```
Comment (harshgoel05) [inline src/utils/toggleSlice.js]:
```suggestion
    setToggleMenu: (state) => {
```

**Review by harshgoel05 — CHANGES_REQUESTED** [review]

File: src/components/Sidebar.js (hunk 9 above)
Comment (vanshika703) [reply src/components/Sidebar.js]: isMenuOpen should be moved?

File: src/components/Sidebar.js (hunk 11 above)
Comment (vanshika703) [reply src/components/Sidebar.js]: not required to center sidebar items

File: src/components/Sidebar.js (hunk 12 above)
Comment (vanshika703) [reply src/components/Sidebar.js]: How?

File: src/utils/toggleSlice.js (hunk 14 above)
Comment (vanshika703) [reply src/utils/toggleSlice.js]: We are toggling the menu, hence, that name.

File: src/components/Sidebar.js (hunk 9 above)
Comment (harshgoel05) [reply src/components/Sidebar.js]: sidebarItems

File: src/components/Sidebar.js (hunk 11 above)
Comment (harshgoel05) [reply src/components/Sidebar.js]: Please provide a screenshot how is it now?

File: src/components/Sidebar.js (hunk 12 above)
Comment (harshgoel05) [reply src/components/Sidebar.js]: sidebarItem.icon => sidebaritem.name 
every image should have a name/alt field in json array

File: src/utils/toggleSlice.js (hunk 14 above)
Comment (harshgoel05) [reply src/utils/toggleSlice.js]: but you are setting the Toggle so we basically follow: gettter/setter function convention

getFormData
setFormData

File: src/components/Body.js [hunk 15]
```diff
@@ -1,5 +1,4 @@
 import React from "react";
```
Comment (harshgoel05) [inline src/components/Body.js]: Remove 
```suggestion

```

**Review by harshgoel05 — CHANGES_REQUESTED** [review]

File: src/components/Header.js (hunk 7 above)
Comment (vanshika703) [reply src/components/Header.js:28]: I have checkedout the article, looks good! However, I would like to continue with the current method for this project as it implements debouncing from scratch.

File: src/components/Sidebar.js (hunk 11 above)
Comment (vanshika703) [reply src/components/Sidebar.js]: image

Comment (vanshika703) [reply src/components/Sidebar.js]: image
Done

**Review by harshgoel05 — APPROVED** [review]

**Pull request merged**